### PR TITLE
Add basic search bar

### PR DESCRIPTION
**Pull request by MrSnoodle01**
# Pull Request Template

## Description

Added basic search bar functionality. You can use the search bar to search for a category, or books depending on which screen you are on
Here are some images of the search bar:

## Type of Change

- [ ] Bugfix
- [x] New Feature
- [ ] Documentation Update
- [ ] Other (please describe):

## Checklist

- [ ] I have read the contributing guidelines.
- [x] I have followed the code style of this project.
- [ ] I have added tests to cover my changes.
- [ ] I have updated the documentation (if applicable).
- [x] My changes do not generate new warnings.

### CODE REVIEW

**Comment by LilAttacker** [comment]
Ok, this has some issues I would like to see fixed before I'll approve this.

Firstly, you added two comments in 381 lines. Please add some more documentation to this. They don't need to be complicated or particularly detailed, but just some simple explanations or even section descriptions saying what's going on or what the purpose is would go a long way.

--

Secondly, you have several warnings indicated by GitHub actions. Please resolve these.

Mostly they seem to be the same error, but I'd like to see them addressed before merging.

--

Thirdly, your description is quite lacking in this PR. You have two sentences describing ~450 lines of changed code. Longer and more detailed descriptions describing your changes (and why you made them!) makes it easier on us to review them (and is also part of the assignment guidelines).

Not only that, you mention your addition of a search bar into the app. This is a good feature but its difficult to judge without visual aides. Adding some screenshots so we can comment or provide feedback to it would be very useful. For example, we may not like how it looks or where its positioned, but its difficult to know this prior to merging.

--

Fourthly, you mention an issue in the additional notes section:

> The search bar on the main categories page is not behaving with the add categories button and isn't styled correctly.

Could you elaborate? Provide an example or screenshot? Can you answer how problematic this is? And more importantly, can you fix it? I'm fine merging even with this issue so long as we know why this issue is there and that there is an expectation for it to be fixed in a future PR (like for example it may be relying on a feature not yet implemented or something).

Others may not be fine merging with this though. So knowing what's causing this could be helpful and others may have you address it before its allowed to be merged..

File: Tabby/app/library/[category]/index.tsx [hunk 1]
```diff
@@ -52,18 +54,32 @@ const CategoryPage: React.FC = () => {
             <FavoriteButtonIcon isFavorite={book.isFavorite} />
         </Pressable>
     );
+    
+    const renderItem = ({item}: {item: Book}) => {
+        console.log("Search: ", search.length);
```
Comment (David1000734) [inline Tabby/app/library/[category]/index.tsx:59]: Can we add comments here just to remind ourselves to remove the log stuff in production build.
A simple 
// DEBUG 
or something would do

**Review by David1000734 — CHANGES_REQUESTED** [review]
Some minor changes needed. GitHub Actions wants you to change the == into === and also add some comments onto the console.log incase we ever want to go back and remove them for prod build

**Review by caesarx26 — COMMENTED** [review]
For the search bar when comparing string use .includes instead of ===

**Review by ederic-oytas — COMMENTED** [review]
Hi Josh, I've reviewed your code and I agree with @LilAttacker about the points he made: (1) needs more comments, (2) needs to address Github actions warnings, and (3 & 4) needs more description. So, please make sure to make those changes with the PR.

Style-wise, I think the code looks good, as the variable and function names remain consistently camelCased. Although, there are some weird spacings like below:
```
            <Pressable
              onPress={() => handleCategoryPress(item)}
              onLongPress={() => handleLongPress(item.name)}
              className={`flex-row items-center justify-between py-4 px-6 
                              ${
                                item.isSelected
                                  ? "bg-blue-500"
                                  : index % 2 === 0
                                  ? "bg-black"
                                  : "bg-gray-300"
                              } opacity-5`}
              style={{ opacity: item.isSelected ? 0.7 : 1 }}
            >
```
But, I think this issue can be addressed in a future PR by introducing a formatter such as Prettier. For now, if you get the 4 changes described earlier, I would approve this PR.

**Pull request merged**